Lesson25/Anil/filter.js

```diff
@@ -0,0 +1,428 @@
+const sentencesExercise5 = [
+	'The quick brown fox jumps over the lazy dog.',
+	'I love to code and solve problems.',
+	'The sun sets in the west.',
+	'Life is like a box of chocolates.',
+	'The cat meowed loudly.',
+	'She walked along the beach, enjoying the salty breeze.',
+	'The rain poured down, soaking everything in sight.',
+	'He smiled and waved goodbye.',
+	'The mountain peaks were covered in snow.',
+	'The room was filled with laughter and joy.',
+	'I have been dreaming of traveling the world and exploring new cultures and cuisines.',
+	'The city lights shimmered in the night sky, creating a mesmerizing spectacle.',
+	'She diligently practiced the piano for hours, striving for perfection in every note.',
+	'The ancient ruins stood as a testament to a long-lost civilization, captivating the imaginations of visitors.',
+	'As the waves crashed against the shore, a sense of tranquility washed over me.',
+	'In the depths of the forest, hidden treasures await those who dare to seek them.',
+	'The starry night sky stretched endlessly, a canvas of beauty and mystery.',
+	'With each step, the hiker felt closer to the summit, fueled by determination and a sense of adventure.',
+	'The aroma of freshly brewed coffee permeated the air, enticing passersby to indulge in a cup of warmth.',
+	'The butterfly gracefully fluttered its wings, showcasing vibrant colors in a delicate dance.',
+];
+
+const under10Words = (array,length) => [...array].filter((sentence)=> sentence.split(" ").length < 10 );
+console.log("under10Words:", under10Words(sentencesExercise5,10) ) ;
+
+
+// Exercise 6:
+// Given an array of products, filter out the products that are out of stock (where the quantity is 0).
+const productsExercise6 = [
+	{
+		name: 'Laptop',
+		brand: 'Apple',
+		price: 1999,
+		quantity: 0,
+	},
+	{
+		name: 'Smartphone',
+		brand: 'Samsung',
+		price: 899,
+		quantity: 20,
+	},
+	{
+		name: 'TV',
+		brand: 'LG',
+		price: 1499,
+		quantity: 15,
+	},
+	{
+		name: 'Headphones',
+		brand: 'Sony',
+		price: 299,
+		quantity: 30,
+	},
+	{
+		name: 'Smartwatch',
+		brand: 'Apple',
+		price: 399,
+		quantity: 25,
+	},
+	{
+		name: 'Camera',
+		brand: 'Canon',
+		price: 999,
+		quantity: 0,
+	},
+	{
+		name: 'Gaming Console',
+		brand: 'Microsoft',
+		price: 499,
+		quantity: 8,
+	},
+	{
+		name: 'Tablet',
+		brand: 'Samsung',
+		price: 649,
+		quantity: 18,
+	},
+	{
+		name: 'Smart Speaker',
+		brand: 'Amazon',
+		price: 99,
+		quantity: 0,
+	},
+	{
+		name: 'Fitness Tracker',
+		brand: 'Fitbit',
+		price: 129,
+		quantity: 14,
+	},
+];
+
+const outOfStock = (array,stock) => [...array].filter((product)=> product.quantity == stock );
+console.log("outOfStock:", outOfStock(productsExercise6,0) ) ;
+
+
+// Exercise 7:
+// Given an array of numbers, filter out the numbers that are divisible by 3.
+const numbersExercise7 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const divisiblebyX = (array,x) => [...array].filter((number)=> number % x == 0 );
```

Comment:
Divisor could be a better name in this situation 

```suggestion
const divisiblebyX = (array, divisor) => [...array].filter((number)=> number % divisor == 0 );
```